mlflow/utils/process.py

```diff
@@ -73,7 +85,16 @@ def _exec_cmd(
             "`capture_output=True` and `stream_output=True` cannot be specified at the same time"
         )
 
-    env = env if extra_env is None else {**os.environ, **extra_env}
+    # Copy current `os.environ` or passed in `env` to avoid mutating it.
+    env = env or os.environ.copy()
+    if extra_env is not None:
+        env.update(extra_env)
```

Comment:
do we need this change?

Reply:
the passed in `env` might be `None`, to ensure `_remove_inaccesible_python_path` gets a valid dict, so I slightly refactor code here.

Reply:
makes sense!